### DISCUSSION TITLE
Buttons not assigning new pub to default collection

### DISCUSSION THREAD

**Discussion by kyledenlinger** [knowledgefutures/pubpub#2908] [Support] [answered]
First I've encountered this issue. It's not consistent across my community so I don't know what's going on.

This morning I created [this page](https://livingquestions.pubpub.org/experiential-learning) and three collections: [Summer](https://livingquestions.pubpub.org/summer), [Study Away](https://livingquestions.pubpub.org/study-away), and [Experiential Learning.](https://livingquestions.pubpub.org/experiential-learning-2). (Notice the page in question and the Experiential Learning collection have the same title.)

At the bottom of that page, I have a banner block with a Create Pub button that I want to default to the Experiential Learning collection. This setting doesn't seem to stick. New pubs created with that button don't get assigned to any collection.

Other similar buttons in the same community work as expected.

**Comment by tefkah** [comment, marked as answer]
Hi, thanks for reporting this bug!

I can reproduce this, and I think I know where the issue lies. We will try to fix it as soon as possible!
Apparently you can only have the default collection be the Collection the banner is on, or the default collections you set in `/dash/settings/pub-settings`.

As a temporary fix, this happens when you put the Create Pub banner on a page, but _not_ when you put that banner in a Collection layout.
So you are able to put this button on the "built in" page for that Collection, rather than creating a separate page for it, it will work.

**Comment by kyledenlinger** [comment]
Oh, that makes sense. Thanks. I just emailed about another button-related issue that a faculty member has reported, and that I've also noticed. I'll share it here.

When not logged in, "Log in to create pub" buttons behave erratically. Sometimes my login state is preserved, other times I'm redirected to the community homepage in a logged-out state, other times (I think) I'm directed to my profile page, not back to the target community. Here's [the most recent community](https://brainsonwriting.pubpub.org/) in which I've noticed this. To be clear, I didn't build this one and haven't spent much time there, so it's possible the faculty member and I are overlooking something simple.

**Reply by tefkah** [reply]
This is most likely a caching issue, which is something we are going to address in the near future!
